pacman/utilities/algorithm_utilities/routing_info_allocator_utilities.py

```diff
@@ -26,39 +32,542 @@ def get_edge_groups(machine_graph, traffic_type):
     """
 
     # Keep a dictionary of the group which contains an edge
-    fixed_key_groups = set()
-    fixed_mask_groups = set()
-    fixed_field_groups = set()
-    flexi_field_groups = set()
-    continuous_groups = set()
+    fixed_key_groups = list()
+    shared_key_groups = list()
+    fixed_mask_groups = list()
+    fixed_field_groups = list()
+    flexi_field_groups = list()
+    continuous_groups = list()
```

Comment:
I'm concerned about using a list instead of a set. Perhaps an OrderedSet would be better?

Reply:
6 of one, 6 of an other. not really sure.

Reply:
Having looked at this, they are now different things; in my changed version it is probably clearer that they are never added to twice in any case